src/main/java/org/sonarlint/intellij/actions/ReviewSecurityHotspotAction.kt

```diff
@@ -0,0 +1,105 @@
+import com.intellij.notification.NotificationType
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.DataKey
+import com.intellij.openapi.module.ModuleUtil
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import org.sonarlint.intellij.analysis.AnalysisStatus
+import org.sonarlint.intellij.common.ui.SonarLintConsole
+import org.sonarlint.intellij.common.util.SonarLintUtils
+import org.sonarlint.intellij.core.BackendService
+import org.sonarlint.intellij.core.ProjectBindingManager
+import org.sonarlint.intellij.finding.hotspot.LiveSecurityHotspot
+import org.sonarlint.intellij.ui.review.ReviewSecurityHotspotDialog
+
+class ReviewSecurityHotspotAction : AbstractSonarAction(
+    "Review Security Hotspot", "Review Security Hotspot Status on SonarQube", AllIcons.Actions.BuildLoadChanges
+) {
+
+    companion object {
+        val GROUP: NotificationGroup = NotificationGroupManager.getInstance().getNotificationGroup("SonarLint: Security Hotspot Review")
+        val SECURITY_HOTSPOT_KEY = DataKey.create<LiveSecurityHotspot>("sonarlint_security_hotspot")
+    }
+
+    override fun isEnabled(e: AnActionEvent, project: Project, status: AnalysisStatus): Boolean {
+        return e.getData(SECURITY_HOTSPOT_KEY) != null && e.getData(SECURITY_HOTSPOT_KEY)?.serverFindingKey != null
+    }
+
+    override fun actionPerformed(e: AnActionEvent) {
+        val project = e.project ?: return
+        val securityHotspot =
+            e.getData(SECURITY_HOTSPOT_KEY) ?: return displayErrorNotification(project, "The security hotspot could not be found.")
+
+        securityHotspot.serverFindingKey?.let {
+            openReviewingDialog(project, it, securityHotspot.file)
+        }
+    }
+
+    fun openReviewingDialog(project: Project, securityHotspotKey: String, file: VirtualFile) {
+        val module = ModuleUtil.findModuleForFile(file, project) ?: return displayErrorNotification(
+            project, "No module could be found for this file."
+        )
+        val connection = SonarLintUtils.getService(project, ProjectBindingManager::class.java).tryGetServerConnection()
+        if (connection.isEmpty) {
+            return displayErrorNotification(project, "No connection to SonarQube could be found.")
+        }
+
+        SonarLintUtils.getService(BackendService::class.java).listAllowedStatusesForHotspots(connection.get().name)
+            .thenAccept { listAllowedStatusesResponse ->
+                val listStatuses = listAllowedStatusesResponse.allowedStatuses
+                if (listStatuses.isEmpty()) {
+                    displayErrorNotification(project, "The statuses for this security hotspot could not be retrieved.")
+                } else {
+                    if (ReviewSecurityHotspotDialog(project, listStatuses, module, securityHotspotKey).showAndGet()) {
+                        displaySuccessfulNotification(project)
```

Comment:
Will the security hotspot tab be automatically updated to remove the resolved hotspot ? @nquinquenel

Reply:
Nope, good point, I forgot to implement this!